NitroxServer/Program.cs

```diff
@@ -58,5 +81,26 @@ static void configureCultureInfo()
             Thread.CurrentThread.CurrentCulture = cultureInfo;
             Thread.CurrentThread.CurrentUICulture = cultureInfo;
         }
+        
+        // See: https://docs.microsoft.com/en-us/windows/console/setconsolectrlhandler
+        static ConsoleEventDelegate consoleHandler;
+        private delegate bool ConsoleEventDelegate(int eventType);
+        [DllImport("kernel32.dll", SetLastError = true)]
+        private static extern bool SetConsoleCtrlHandler(ConsoleEventDelegate callback, bool add);
+        private static bool ConsoleEventCallback(int eventType)
+        {
+            if (eventType == 2)
+            {
+                Log.Info("Exiting ...");
+                Server.Instance.Stop();
+            }
+            return false;
+        }
+        private static void OnCtrlCPressed(object sender, ConsoleCancelEventArgs e)
+        {
+            Log.Info("Exiting ...");
```

Comment:
Logging this isn't needed because the server already does it.

Reply:
Saving takes a few seconds for me running it on GCP, this is to notify users that their cancel request is received.